contracts/L2Base.sol

```diff
@@ -27,9 +28,9 @@ interface ISoulGasToken {
 }
 
 /// @custom:proxied
-/// @title EthStorageContractL2
-/// @notice EthStorage contract that will be deployed on L2, and uses L1Block contract to mine.
-contract EthStorageContractL2 is EthStorageContract2 {
+/// @title L2Base
+/// @notice Common base contract that will be deployed on L2, and uses L1Block contract to mine.
+abstract contract L2Base is AccessControlUpgradeable {
```

Comment:
We could make `setSoulGasToken` internal, and let EthStorageContract add the access control